### ISSUE TITLE
steps: Step9DreamingState.step_count wraps negative at int32 max

### ISSUE THREAD

**Issue by rama0x1** [SlopDotCash/asi#1905]
## Summary

\`step9_update\` in \`alberta_framework/steps/step9.py\` advances the Step 9
guarded-dreaming facade's own lifetime clock with a bare int32 increment:

\`\`\`python
new_state = Step9DreamingState(
    ...
    step_count=state.step_count + 1,
)
\`\`\`

\`Step9DreamingState.step_count\` is a \`jnp.int32\` array (initialized via
\`jnp.array(0, dtype=jnp.int32)\` in \`init_step9_state\`). JAX/NumPy int32
addition wraps silently on overflow: at \`INT32_MAX\` (\`2**31 - 1\`), \`+ 1\`
produces \`INT32_MIN\` (\`-2**31\`), not a saturated or raised value.

This is the same lifetime-clock defect already fixed for
\`alberta_framework.streams.alberta_plan_step1\` (#1369),
\`alberta_framework.streams.feature_discovery\` (#1367),
\`alberta_framework.streams.out_of_class\` (#1434), and
\`alberta_framework.streams.synthetic\` (#1365) via
\`alberta_framework.core.normalizers._saturating_int32_counter_increment\`,
and matches the still-open siblings tracked in #1787 (MixedHorde), #1841
(UPGDLearner), and #1843 (GauntletStream).

## Impact

\`Step9DreamingState\` is part of the public Step 9 library surface (exported
from \`alberta_framework.steps\`). External callers that treat
\`state.step_count\` as a monotonically non-negative counter (checkpoint
cadence, logging, external telemetry) get a corrupted value after ~2.1
billion real transitions, with no error raised.

Note: this facade-level counter is separate from
\`model_state.step_count\` (the world model's own internal counter, gated at
\`config.dreaming_warmup_steps\`), which is a different field on a different
object and is not affected by this issue.

## Fix

Replace the bare \`+ 1\` with
\`alberta_framework.core.normalizers._saturating_int32_counter_increment\`,
matching the existing pattern in the streams modules above.

## Related, out of scope here

\`alberta_framework/steps/step7.py\`'s \`Step7DynaState.step_count\` has the
identical bare-\`+ 1\` pattern at its own \`step7_update\` return site. Left
unfixed in this issue/PR to keep the change to one file; worth a follow-up
issue if nobody has picked it up.
